src/mongo_logger.ts

```diff
@@ -119,61 +189,273 @@ function parseSeverityFromString(s?: string): SeverityLevel | null {
  */
 function resolveLogPath(
   { MONGODB_LOG_PATH }: MongoLoggerEnvOptions,
-  {
-    mongodbLogPath
-  }: {
-    mongodbLogPath?: unknown;
+  { mongodbLogPath }: ClientLogPathOptions
+): MongoDBLogWritable {
+  if (typeof mongodbLogPath === 'string' && /^stderr$/i.test(mongodbLogPath))
+    return createStdLogger(process.stderr);
+  if (typeof mongodbLogPath === 'string' && /^stdout$/i.test(mongodbLogPath))
+    return createStdLogger(process.stdout);
+
+  if (typeof mongodbLogPath === 'object' && typeof mongodbLogPath?.write === 'function') {
+    return mongodbLogPath;
   }
-): Writable {
-  const isValidLogDestinationString = (destination: string) =>
-    ['stdout', 'stderr'].includes(destination.toLowerCase());
-  if (typeof mongodbLogPath === 'string' && isValidLogDestinationString(mongodbLogPath)) {
-    return mongodbLogPath.toLowerCase() === 'stderr' ? process.stderr : process.stdout;
+
+  if (MONGODB_LOG_PATH && /^stderr$/i.test(MONGODB_LOG_PATH)) {
+    return createStdLogger(process.stderr);
+  }
+  if (MONGODB_LOG_PATH && /^stdout$/i.test(MONGODB_LOG_PATH)) {
+    return createStdLogger(process.stdout);
   }
 
-  // TODO(NODE-4813): check for minimal interface instead of instanceof Writable
-  if (typeof mongodbLogPath === 'object' && mongodbLogPath instanceof Writable) {
-    return mongodbLogPath;
+  return createStdLogger(process.stderr);
+}
+
+/** @internal */
+export interface Log extends Record<string, any> {
+  t: Date;
+  c: MongoLoggableComponent;
+  s: SeverityLevel;
+  message?: string;
+}
+
+/** @internal */
+export interface MongoDBLogWritable {
+  write(log: Log): unknown;
```

Comment:
Good catch. When deciding on what the `MongoDBLogWritable` interface should look like, I wasn't sure if we wanted the write method to have a return value that would indicate success or failure that we'd have to handle inside the private log method, but I think we're good to change this to return `void`.

Reply:
This type is internal for now but the intention is that it would be put on `MongoClientOptions` right?

I think `unknown` is correct here, `void` will prevent APIs that have a return value which is not a requirement we need to impose.

Reply:
Yea because it would not matter to us I don't think we should have the enforcement of a return value